### PR TITLE
Add internal solution for ceoi-11-bal (Balloons) with cpp and java code [ceoi11]

### PR DESCRIPTION
**Pull request by ChuangSheep**
_If any of the below doesn't (?) apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.
  
  
  Judge Result:
https://oj.uz/submission/604111
https://oj.uz/submission/604103

### CODE REVIEW

File: solutions/ceoi-11-bal.mdx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+<LanguageSection>
+<CPPSection>
+## Implementation
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using ll = long long;
+using ld = long double;
```
Comment (SansPapyrus683) [inline solutions/ceoi-11-bal.mdx:29]: try just using `double`?

File: solutions/ceoi-11-bal.mdx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+		 * save the x coordinate and radius of the current balloon, so that
+		 * it can be checked later when a new balloon being inflated
+		 */
+		to_check.push({x, max_r});
+
+		res[i] = max_r;
+	}
+
+	cout << fixed << setprecision(3);
+	for (auto &r : res)
```
Comment (SansPapyrus683) [inline solutions/ceoi-11-bal.mdx:110]: i wouldn't use `auto` here

File: solutions/ceoi-11-bal.mdx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+				maxR = Math.min(maxR, toLastR);
+
+				/*
+				 * if current maximum radius >= radius of the last balloon, we can
+				 * remove the last balloon since it will never be touched by any
+				 * new balloons other than the current one
+				 */
+				if (maxR >= toCheck.peek().r) {
+					toCheck.pop();
+					// check the next balloon saved which will possibly reduce max_r
```
Comment (SansPapyrus683) [inline solutions/ceoi-11-bal.mdx:171]: over 80 chars?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Comment by SansPapyrus683** [comment]
also why different brace styles?

File: solutions/ceoi-11-bal.mdx (hunk 3 above)
Comment (ChuangSheep) [reply solutions/ceoi-11-bal.mdx:171]: actually only 75, but i could split it to two lines if preferred

Comment (SansPapyrus683) [reply solutions/ceoi-11-bal.mdx:171]: yeah nvm

File: solutions/ceoi-11-bal.mdx [hunk 4]
```diff
@@ -107,8 +98,7 @@ int main()
 	}
 
 	cout << fixed << setprecision(3);
```
Comment (SansPapyrus683) [inline solutions/ceoi-11-bal.mdx:100]: put `3` in a constant called `PRECISION`

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
@envyaims

File: solutions/ceoi-11-bal.mdx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+source: CEOI 2011
+title: Balloons
+author: Chuyang Wang
+---
+
+[Official Analysis](https://oi.edu.pl/static/attachment/20110713/ceoi-2011.pdf#page=11)
+
+## Explanation
+
+The problem can be solved by simulation. In particular, we want to inflate the balloons from left to right. The difficulty is to find out when the new balloon $b$ touches one of the inflated balloon $a$ on its left, and how to check this for all previous inflated balloons efficiently. For notation, $a_r$ is the radius and $a_x$ the x-coordinate of balloon $a$.
```
Comment (envyaims) [inline solutions/ceoi-11-bal.mdx:12]:
```suggestion
The problem can be solved by simulation. In particular, we want to inflate the balloons from left to right. The difficult part is to find out when the new balloon $b$ touches one of the inflated balloon $a$ on its left, and how to check this for all previous inflated balloons efficiently. For notation, $a_r$ is the radius and $a_x$ the x-coordinate of balloon $a$. 
```
?

File: solutions/ceoi-11-bal.mdx [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+ * how long can the radius of the new balloon at position bx be
+ * so that it touches the ballon a, which is described by
+ * its x position a.first and its radius a.second
+ */
+double calc_r(pair<double, double> a, double bx) {
+	return (a.first - bx) * (a.first - bx) / (4 * a.second);
+}
+
+int main() {
+	// number of balloons
```
Comment (envyaims) [inline solutions/ceoi-11-bal.mdx:40]: dont state the obvious yo

File: solutions/ceoi-11-bal.mdx [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+	// the radius of each balloon after inflating
+	vector<double> res(n);
+	// the balloons we should check when we add a new one
+	stack<pair<double, double>> to_check;
+
+	// simulate inflating balloons
+	for (int i = 0; i < n; i++) {
+		/*
+		 * the x coordinate and the maximum radius of the
```
Comment (envyaims) [inline solutions/ceoi-11-bal.mdx:52]: unnecessary comment

File: solutions/ceoi-11-bal.mdx [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+	return (a.first - bx) * (a.first - bx) / (4 * a.second);
+}
+
+int main() {
+	// number of balloons
+	int n;
+	cin >> n;
+
+	// the radius of each balloon after inflating
+	vector<double> res(n);
```
Comment (envyaims) [inline solutions/ceoi-11-bal.mdx:45]: maybe a better variable name than `res`?

**Review by envyaims — CHANGES_REQUESTED** [review]
great stuff!
also remember to apply the same changes to java code

**Review by envyaims — APPROVED** [review]
very nice :blobheart:

**Pull request merged**